src/libs/AxelarQueryAPI.ts

```diff
@@ -309,30 +341,70 @@ export class AxelarQueryAPI {
       sourceToken.decimals
     );
 
-    const executionFee = destGasFeeWei.gt(minDestGasFeeWei)
+    const excludedL1ExecutionFee = destGasFeeWei.gt(minDestGasFeeWei)
       ? srcGasFeeWei
       : srcGasFeeWei.mul(minDestGasFeeWei).div(destGasFeeWei);
 
     const actualGasMultiplier = gasMultiplier === "auto" ? executeGasMultiplier : gasMultiplier;
 
-    const executionFeeWithMultiplier =
+    const excludedL1ExecutionFeeWithMultiplier =
       actualGasMultiplier > 1
-        ? executionFee.mul(actualGasMultiplier * 10000).div(10000)
-        : executionFee;
+        ? excludedL1ExecutionFee.mul(actualGasMultiplier * 10000).div(10000)
+        : excludedL1ExecutionFee;
+
+    let l1ExecutionFee = BigNumber.from(0);
+    let l1ExecutionFeeWithMultiplier = BigNumber.from(0);
+
+    // If the destination chain is L2, calculate the L1 execution fee
+    if (isL2Chain(destinationChainId)) {
+      if (!executeData) {
+        throw new Error(
```

Comment:
if they don't pass in the executeData, they may just be invoking it to get a sense of what the actual fee is and not have a specific payload in mind. 

i'm wondering if it's better to just show a warning instead of throwing an error here, something like `Looks like you are trying to get a gas estimate for execution on an L2 chain. To more accurately measure the cost of this cross-chain call, the cost of posting your transaction to the L1 should be factored in. This method can calculate that for you if you pass in the intended payload for it, without it, consider either bumping up your gasLimit or gasMultiplier parameters to include a buffer for the L1 inclusion fee`.

Reply:
IMO, throwing an error instead of just warning will force dev to implement it correctly. Otherwise, if they use it incorrectly, the issue will be hidden until their users find out which i think is worse.

Reply:
ah, i just saw your response to my comment after i submitted #300. i think we can just add a warning, especially since we don't even need the executeData in all cases (e.g. in cases of arbitrum where it's already included in the l2)